http/wrap_writer.go

```diff
@@ -0,0 +1,178 @@
+package sentryhttp
+
+import (
+	"bufio"
+	"io"
+	"net"
+	"net/http"
+)
+
+// This wrapper is derived from https://github.com/go-chi/chi/blob/master/middleware/wrap_writer.go
```

Comment:
Please include the entire content of their license when copying/deriving code.
https://github.com/go-chi/chi/blob/master/LICENSE